scripts/extract_pymatgen_data.py

```
@@ -0,0 +1,175 @@
+        with gzip.open(path, "rb") as file:
+            return json.loads(file.read())
+    if path.suffix == ".json":
+        with open(path, encoding="utf-8") as file:
+            return json.load(file)
+
+    # For TypeScript files, use Deno to export as JSON
+    if not shutil.which("deno"):
+        raise RuntimeError(
+            "deno is required to parse TypeScript files. "
+            "Install from https://deno.land or run: curl -fsSL https://deno.land/install.sh | sh"
+        )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)